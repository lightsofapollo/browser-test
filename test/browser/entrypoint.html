<!DOCTYPE HTML>
<html>
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8" />
  <title>body</title>
  <link rel="stylesheet" href="/node_modules/mocha/mocha.css" title="" type="text/css" />
</head>
<body>
<div id="mocha"></div>
<script src="/browser_test/browser_test.js" type="text/javascript" charset="utf-8"></script>
<script src="/node_modules/mocha/mocha.js" type="text/javascript" charset="utf-8"></script>
<script type="text/javascript" charset="utf-8">

function Reporter(runner) {
  runner.on('start', function(evt) {
    browserTest.createSuite(this.total, 'mocha.html');
  });

  runner.on('test', function(evt) {
    browserTest.createTest(evt.fullTitle());
  });

  runner.on('pass', function(evt) {
    browserTest.passTest();
  });

  runner.on('fail', function(evt, err) {
    browserTest.failTest(err);
  });

  runner.on('pending', function(evt) {
    browserTest.createTest(evt.title)
    browserTest.pendingTest();
  });

  runner.once('end', function() {
    browserTest.finishSuite();
  });
}

mocha.setup({
  ui: 'tdd',
  reporter: Reporter
});

</script>

<script type="text/javascript" charset="utf-8">

// create the requested test
var script = document.createElement('script');
// XXX: We need some method to fail the test if load fails
script.src = browserTest.fileName;
script.onerror = function(err) {
  console.error('test load failure!', err);
};

script.onload = function() {
  mocha.run();
};
document.body.appendChild(script);
</script>
</body>
</html>



