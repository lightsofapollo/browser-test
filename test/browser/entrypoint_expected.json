{
  "ok": false,
  "asserts": [
    {
      "ok": true,
      "number": 1,
      "name": "mocha js test woot"
    },
    {
      "ok": false,
      "number": 2,
      "name": "mocha js test fail"
    }
  ],
  "pass": [
    {
      "ok": true,
      "number": 1,
      "name": "mocha js test woot"
    }
  ],
  "fail": [
    {
      "ok": false,
      "number": 2,
      "name": "mocha js test fail"
    }
  ],
  "todo": [],
  "errors": [],
  "plan": {
    "start": 1,
    "end": 2
  }
}